src/py_converters.cpp

```diff
@@ -256,18 +255,15 @@ int convert_dashes(PyObject *dashobj, void *dashesp)
         return 0;
     }
 
-    nentries = PySequence_Size(dashes_seq);
-    if (nentries % 2 != 0) {
-        PyErr_Format(PyExc_ValueError, "dashes sequence must have an even number of elements");
-        return 0;
-    }
+    Py_ssize_t nentries = PySequence_Size(dashes_seq);
+    Py_ssize_t dash_pattern_length = (nentries % 2) ? 2 * nentries : nentries;
```

Comment:
Can you add a comment what this is doing and why? It's hard to understand if one doesn't know the context.

Reply:
done